PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx

```diff
@@ -0,0 +1,388 @@
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kDCAZ]), track.dcaZ());
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kDCAXY]), track.dcaXY());
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCLUSTERS]), track.tpcNClsFound());
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCROSSEDROWS]), track.tpcNClsCrossedRows());
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kTPCCHI2]), track.tpcChi2NCl());
+    registry.fill(HIST(AR::TCrecoAfter) + HIST(AR::TrackVariableNames[AR::kITSCLUSTERS]), track.itsNCls());
+  }
+  template <typename TrackInstance>
+  bool SurviveTrackCut(TrackInstance const& track)
+  {
+    bool SurviveCut = true;
+
+    // cut pt
+    if (cfgTrack.at(AR::kPT).value.at(5) > 0 &&
+        (track.pt() < cfgTrack.at(AR::kPT).value.at(3) || track.pt() > cfgTrack.at(AR::kPT).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kETA).value.at(5) > 0 &&
+        (track.eta() < cfgTrack.at(AR::kETA).value.at(3) || track.eta() > cfgTrack.at(AR::kETA).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kPHI).value.at(5) > 0 &&
+        (track.phi() < cfgTrack.at(AR::kPHI).value.at(3) || track.phi() > cfgTrack.at(AR::kPHI).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kCHARGE).value.at(5) > 0 &&
+        (track.sign() < cfgTrack.at(AR::kCHARGE).value.at(3) || track.sign() > cfgTrack.at(AR::kCHARGE).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kDCAZ).value.at(5) > 0 &&
+        (track.dcaZ() < cfgTrack.at(AR::kDCAZ).value.at(3) || track.dcaZ() > cfgTrack.at(AR::kDCAZ).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kDCAXY).value.at(5) > 0 &&
+        (track.dcaXY() < cfgTrack.at(AR::kDCAXY).value.at(3) || track.dcaXY() > cfgTrack.at(AR::kDCAXY).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kTPCCLUSTERS).value.at(5) > 0 &&
+        (track.tpcNClsFound() < cfgTrack.at(AR::kTPCCLUSTERS).value.at(3) || track.tpcNClsFound() > cfgTrack.at(AR::kTPCCLUSTERS).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kTPCCROSSEDROWS).value.at(5) > 0 &&
+        (track.tpcNClsCrossedRows() < cfgTrack.at(AR::kTPCCROSSEDROWS).value.at(3) || track.tpcNClsCrossedRows() > cfgTrack.at(AR::kTPCCROSSEDROWS).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kTPCCHI2).value.at(5) > 0 &&
+        (track.tpcChi2NCl() < cfgTrack.at(AR::kTPCCHI2).value.at(3) || track.tpcChi2NCl() > cfgTrack.at(AR::kTPCCHI2).value.at(4))) {
+      SurviveCut = false;
+    }
+    if (cfgTrack.at(AR::kITSCLUSTERS).value.at(5) > 0 &&
+        (track.itsNCls() < cfgTrack.at(AR::kITSCLUSTERS).value.at(3) || track.itsNCls() > cfgTrack.at(AR::kITSCLUSTERS).value.at(4))) {
+      SurviveCut = false;
+    }
+
+    return SurviveCut;
+  }
+
```

Comment:
You should be able to have this in a cleaner way by using a lambda function
Check `ProcessRun3` in https://github.com/AliceO2Group/O2Physics/blob/master/Common/TableProducer/centralityTable.cxx for inspiration